pkg/processor/transform/txfjs/transformer_test.go

```diff
@@ -160,3 +235,153 @@ function transform(record) {
 		})
 	}
 }
+
+func TestTransformer_DataTypes(t *testing.T) {
+	testCases := []struct {
+		name  string
+		src   string
+		input record.Record
+		want  record.Record
+	}{
+		{
+			name: "UTC date is used",
+			src: `function transform(record) {
+        		record.CreatedAt = new Date(Date.UTC(2021, 0, 2, 3, 4, 5, 6)).toISOString();
+				return record;
+			}`,
+			input: record.Record{},
+			want: record.Record{
+				CreatedAt: time.Date(2021, time.January, 2, 3, 4, 5, 6000000, time.UTC),
+			},
+		},
+		{
+			name: "position from string",
+			src: `function transform(record) {
+        		record.Position = "foobar";
```

Comment:
Nitpick: we are mixing tabs and spaces, since GitHub uses 8 spaces for a tab it looks a bit messed up here.

Reply:
Yup definitely messed up (and ugly).: )